### PR TITLE
openmw-tes3mp: fix build

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Fix build after #68029

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gnidorah** [comment]
@GrahamcOfBorg build openmw-tes3mp

**Comment by gnidorah** [comment]
@GrahamcOfBorg build openmw-tes3mp

**Comment by aanderse** [comment]
@gnidorah are the`gcc 9` bugs being tracked upstream? Are there many?

PS. new `openscenegraph` will be out soon but minor release so shouldn't be impacting. I might ping you in https://github.com/NixOS/nixpkgs/pull/77799 for testing, if you don't mind.

**Comment by gnidorah** [comment]
@aanderse 
> are thegcc 9 bugs being tracked upstream? Are there many?

There are are errors:
/build/source/./apps/openmw-mp/Script/ScriptFunctions.hpp:153:5: error: 'constexpr ScriptFunctionPointer::ScriptFunctionPointer(Function<R, Types ...>) [with R = void; Types = {bool}; Function<R, Types ...> = void (*)(bool)]' called in a constant expression
                                                                        
They are not reported as I don't know the language. Maybe project author will do something.

> PS. new openscenegraph will be out soon but minor release so shouldn't be impacting. I might ping you in #77799 for testing, if you don't mind.

Not necessary. @abbradar overrides `osc` source point in openmw derivation https://github.com/NixOS/nixpkgs/blob/master/pkgs/games/openmw/default.nix#L5 and I just reuse openmw package.

**Comment by aanderse** [comment]
@gnidorah right I forgot about the custom `openscenegraph` fork.

If you have the motivation to report `gcc 9` errors upstream I'm sure it would be appreciated. I don't think you need to understand every error you report on `github` projects :man_shrugging: 

But yeah, I think this change looks reasonable until those `gcc 9` errors are fixed :+1:

File: pkgs/games/openmw/tes3mp.nix [hunk 1]
```diff
@@ -71,7 +71,7 @@ in openmw.overrideAttrs (oldAttrs: rec {
     description = "Multiplayer for TES3:Morrowind based on OpenMW";
     homepage = https://tes3mp.com/;
     license = licenses.gpl3;
-    platforms = platforms.linux;
+    platforms = [ "x86_64-linux" "i686-linux" ];
```
Comment (lheckemann) [inline pkgs/games/openmw/tes3mp.nix]: Why this? Is it broken on aarch64-linux?

Comment (gnidorah) [reply pkgs/games/openmw/tes3mp.nix]: @lheckemann Yes it is. This software uses sys/io.h

**Pull request merged**